### DISCUSSION TITLE
(Web3Modal) How to get the selected network when user is not connected

### DISCUSSION THREAD

**Discussion by rafael-farcher** [WalletConnect/walletconnect-monorepo#2320] [Developer Support] [answered]
I am developing a Dapp (React/NextJs) and would like to make it usable even when there is no wallet connection. Currently, when using const `{ chain } = useNetwork();`, the chain is only defined when the user is connected to the network. I would like to access the selected chain state differently to make this possible, but I am unsure how to proceed.

I have tried to access the Network name programmatically but this didn't work for me either since the network switch is a shadow element.

Can you please provide guidance or suggest an alternative method for accessing the selected chain state without requiring the user to be connected to the network? This will greatly improve the user experience of the Dapp by allowing it to be interactable even when not connected.

Thank you for your assistance.

**Comment by 0xAlexander** [comment]
Hey @0xAsimetriq We could really use your help to tackle this issue and keep our project on track. Your quick advice or solution would be greatly appreciated! 🙏

**Comment by finessevanes** [comment]
Going to ping the team - thank you for your patience!

**Comment by xzilja** [comment, marked as answer]
This is coming with new web3modal feature (events), but there is no concrete date for it at the moment. Likely this month however

**Reply by rafael-farcher** [reply]
Thank you for your prompt response! We are eagerly awaiting the new features. In the meantime, we've taken the initiative to implement our own workaround.
